docs/utils/masked/index.rst

```diff
@@ -0,0 +1,165 @@
+.. |Masked| replace:: :class:`~astropy.utils.masked.Masked`
+.. |Quantity| replace:: :class:`~astropy.units.Quantity`
+.. _utils-masked:
+
+**************************************
+Masked Values (`astropy.utils.masked`)
+**************************************
+
+Often, data sets are incomplete or corrupted and it would be handy to be able
+to mask certain values.  Astropy provides a |Masked| class to help represent
+such data sets.
+
+.. warning:: |Masked| is experimental! While we hope basic usage will remain
+   similar, we are not yet sure whether it will not be necessary to change it
+   to make things work throughout Astropy. This also means that comments and
+   suggestions for improvements are especially welcome!
+
+.. note:: |Masked| is similar to Numpy's :class:`~numpy.ma.MaskedArray`,
+   but it supports subclasses much better and also has some important
+   :ref:`differences in behaviour <utils-masked-vs-numpy-maskedarray>`.
+
+Usage
+=====
+
+Astropy |Masked| instances behave like `~numpy.ndarray` or subclasses such as
+|Quantity| but with a mask associated, which is propagated in operations such
+as addition, etc.::
+
+  >>> import numpy as np
+  >>> from astropy import units as u
+  >>> from astropy.utils.masked import Masked
+  >>> ma = Masked([1., 2., 3.], mask=[False, False, True])
+  >>> ma
+  MaskedNDArray([1., 2., ——])
+  >>> mq = ma * u.m
+  >>> mq + 25 * u.cm
+  <MaskedQuantity [1.25, 2.25,  ———] m>
+
+You can get the values without the mask using
+`~astropy.utils.masked.Masked.unmasked`, or, if you need to control what
+should be substituted for any masked values, with
+:meth:`~astropy.utils.masked.Masked.unmask`::
+
+  >>> mq.unmasked
```

Comment:
Coming back to the name of this attribute, I suspect I'm not the only one that will yearn for the `np.ma` consistency of calling it `data`. In particular I do find this a bit surprising:
```
In [23]: ma = Masked(data=[1,2], mask=[True, False])

In [24]: ma.data
Out[24]: <memory at 0x7ff37059a940>
```
I understand this is just what `np.ndarray` is exposing, but still, the `data` / `mask` duo is easy to explain and has a long heritage.

I suppose there is a question of whether to purposely diverge from the np.ma.MaskedArray API given the behavior differences.

Reply:
It was indeed a very conscious choice to make the new attributes and methods extremely limited and ensure that they always include "mask" in their name - the fact that `Column.data` exists and the resulting problems for `MaskedColumn` are a prime inspiration!

But I think the argument becomes clearer if you look at other classes. E.g., for an ndarray subclass like `MaskedAngle`, does it really makes sense for `angle.data` to be an `Angle`? `angle.unmasked` would be more informative.

And for container classes like `Time` and `SkyCoord`, I'd hope to give those attributes and methods with similar names (following what you did already in `Time`), i.e., they should gain at least `.mask` and perhaps also `.unmask()` and `.unmasked`, which will be clearer than `.data` (in fact, it cannot be `.data` for `SkyCoord`, as that already returns the representation - surely, we'd want it to return the masked representation for a masked SkyCoord).

Reply:
Fair enough!